addons/hr_expense/models/hr_expense.py

```diff
@@ -822,6 +823,12 @@ def _compute_can_reset(self):
         for sheet in self:
             sheet.can_reset = is_expense_user if is_expense_user else sheet.employee_id.user_id == self.env.user
 
+    def _compute_can_approve(self):
```

Comment:
You should decorate this method with `@api.depends_context('uid')` as it's related to the current user